### PR TITLE
adding interactivity to answer reveal

### PR DESCRIPTION
**Pull request by Guntrisoft**
- non-js shows question and answer with no trigger
- trigger uses dough collapse to reveal answer
- transitions on plus icon to convert to minus when open
- overridden styles for call out to MAS green if URL is a MAS one

### CODE REVIEW

**Comment by baphled** [comment]
@Guntrisoft this is still lingering, do we still want to merge this?

**Comment by Guntrisoft** [comment]
@baphled in general the code is ok, and I've been keeping it as close to master as possible, but it's not been merged because a) the answer/reveal callout was put down the priority list and b) it's not quite ready as Jade has some feedback on it a while back that hasn't been implemented.

Not sure what to do :)

**Comment by baphled** [comment]
ahhh, maybe we should chase it up.

**Comment by Guntrisoft** [comment]
@benbarnett @darokel This is probably complete enough to be reviewed and potentially merged in. They'll be able to create an question with basic answer in text and include a text/url for the callout at the bottom of the answer.

File: lib/publify_textfilter_answer_reveal.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        answer = answer.gsub(/^/,"<p>").gsub(/\\n/,"</p><p>").gsub(/$/, "</p>")
+
+        callout_id = Digest::SHA256.hexdigest answer
+
+        "<div class=\"collapsible\">
+          <h3 class=\"collapsible__heading\">#{question}</h3>
+          <div class=\"collapsible__reveal\" data-dough-collapsable-target=\"answer_reveal_#{callout_id}\">
+            #{answer}
+
+            <hr class=\"snippet\" data-snippet-name=\"inlinecallout\" data-snippet-url=\"#{callout_url}\" data-snippet-text=\"#{callout_text}\" />
```
Comment (Guntrisoft) [inline lib/publify_textfilter_answer_reveal.rb]: I'm loving that this worked(!) - a nested text filter - i.e. callout within a callout

**Comment by darokel** [comment]
Just a few comments but otherwise you bossed it :ship:

**Comment by Guntrisoft** [comment]
@darokel have altered things based on your feedback
@benbarnett :+1: ?

**Comment by darokel** [comment]
🚢

File: app/views/shared/_collapsible.html.erb [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
-      <circle fill="#0076AB" cx="50" cy="50" r="36"/>
-      <line fill="none" stroke="#FFFFFF" stroke-width="6" stroke-miterlimit="10" x1="50" y1="35" x2="50" y2="65"/>
-      <line fill="none" stroke="#FFFFFF" stroke-width="6" stroke-miterlimit="10" x1="35" y1="50" x2="65" y2="50"/>
+  <div class="collapsible__reveal" data-dough-collapsable-target="answer_reveal">
+    <p>Bacon ipsum dolor amet short ribs bresaola pig venison kielbasa corned beef swine boudin ball tip ribeye tail porchetta sirloin alcatra. Fatback frankfurter brisket, kevin doner biltong cupim cow sausage ham chuck. Shankle turkey ball tip, bacon picanha short loin pork loin spare ribs frankfurter beef ribs. Filet mignon andouille strip steak pork loin cupim capicola bresaola biltong chuck bacon. Shank ground round fatback porchetta shankle sirloin andouille kevin filet mignon pork belly short loin short ribs spare ribs cow drumstick. Doner cow pig andouille pork, salami tri-tip pork belly leberkas brisket.</p>
```
Comment (benbarnett) [inline app/views/shared/_collapsible.html.erb]: Should this still be here? :smiling_imp:

File: app/views/admin/content/_form.html.erb [hunk 3]
```diff
@@ -259,3 +262,22 @@
...
+<div class="modal fade" id="answerReveal" tabindex="-1" role="dialog" aria-labelledby="answerReveal" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h2>Answer/Reveal snippet</h2>
+      </div>
+      <div class="modal-body">
+        <p>Copy the snippet below and paste it to the desired place by clicking on the 'Source' button. <b>Note:</b> Add a \n to the answer to start a new paragraph.</p>
+        <pre id="ma_says--snippet-container">
+&lt;hr class=&quot;snippet&quot; data-snippet-name=&quot;answerreveal&quot; data-snippet-text-question=&quot;<span style="background:red;color:#fff;">[ENTER QUESTION TEXT HERE]</span>&quot; data-snippet-text-answer=&quot;<span style="background:red;color:#fff;">[ENTER ANSWER TEXT HERE]</span>&quot; data-snippet-text-callout=&quot;<span style="background:red;color:#fff;">[ENTER CALLOUT TEXT HERE]</span>&quot; data-snippet-url-callout=&quot;<span style="background:red;color:#fff;">[ENTER CALLOUT URL HERE]</span>&quot; /&gt;
```
Comment (benbarnett) [inline app/views/admin/content/_form.html.erb]: Do we still need these entities escaped? I thought within a `<pre>` you don't need to?

File: app/views/shared/_collapsible.html.erb (hunk 2 above)
Comment (Guntrisoft) [inline app/views/shared/_collapsible.html.erb]: That shared component is only there for the styleguide - the actual implementation of the module is elsewhere in the ruby as a text filter module

Comment (benbarnett) [inline app/views/shared/_collapsible.html.erb]: Cool

**Comment by benbarnett** [comment]
Looks good :+1: 

Would be nice to minimise the duplication of the markup, but I know this is the same for all these snippet things.

File: app/views/admin/content/_form.html.erb (hunk 3 above)
Comment (Guntrisoft) [inline app/views/admin/content/_form.html.erb]: xmp seems to be only one which doesn't need escaping however its a deprecated tag.

The suggestion of a textarea which is disabled seems like a good idea, if it means we get rid of horrible escaped tags in the markup.

Comment (Guntrisoft) [inline app/views/admin/content/_form.html.erb]: The only minor thing about this is there won't be any highlighted items in RED displayed

**Comment by benbarnett** [comment]
Looks good! :+1:

**Pull request merged**